django_date_extensions/fields.py

```diff
@@ -26,8 +26,13 @@
 class ApproximateDate(object):
     """A date object that accepts 0 for month or day to mean we don't
        know when it is within that month/year."""
-    def __init__(self, year=0, month=0, day=0, future=False, past=False, prefix=None):
-        if future and past:
+    def __init__(self, year=0, month=0, day=0, future=False, past=False, prefix=None, string_format=None):
+        if string_format:
+            if year or month or day or future or past or prefix:
+                raise ValueError("String format can have no year, month, day, prefix, future or past")
+            elif string_format.lower() not in settings.STRING_FORMATS:
```

Comment:
I think it should either only allow precise case, or store lowercased version. Otherwise it is going to be confusing querying on stored values if you don't remember to search case-insensitively. I think only allowing precise case is simpler and clearer.